### ISSUE TITLE
Unable to resolve path to module 'apollo-cache-inmemory'

### ISSUE THREAD

**Issue by jay1337** [benmosher/eslint-plugin-import#975]
I get a warning with `apollo-cache-inmemory` and `apollo-utilities` (which are correctly installed and code runs without any error).

This a React Native project developed on Windows with Atom.

eslint@4.12.0
eslint-plugin-import@2.8.0

**Comment by nevir** [comment]
Same happens for `apollo-cache`.  The main thing they all have in common is that they are built into UMD bundles

**Comment by mattfysh** [comment]
I'm using these & related libraries, however only getting this error for `apollo-cache-inmemory` and `apollo-utilities` out of this list:

```
"apollo-cache-inmemory": "^1.1.1",
"apollo-client": "^2.0.3",
"apollo-link": "^1.0.3",
"apollo-link-http": "^1.2.0",
"apollo-utilities": "^1.0.2",
"graphql": "^0.11.7",
"graphql-tag": "^2.5.0",
"graphql-tools": "^2.7.2",
"react-apollo": "^2.0.1",
```

**Comment by craigbilner** [comment]
The way the `package.json` `main` is written it's incorrectly resolving the path here:

`https://github.com/browserify/resolve/blob/master/lib/sync.js#L73`

For example:

`x` is `/node_modules/apollo-cache-inmemory` and `pkg.main` is `/lib/index.js` which resolves to `/lib/index.js` which ultimately can't be found rather than returning `/node_modules/apollo-cache-inmemory/lib/index.js`

If it's acceptable to write a package main like this, then I guess the bug should be raised with the `resolve` lib?

**Comment by ljharb** [comment]
I'd file a bug with apollo to add the leading dot.

**Comment by craigbilner** [comment]
`https://github.com/browserify/resolve/blob/master/lib/sync.js#L73` is where the package is resolved. `main` is taken from `jsnext:main` which had a typo so didn't resolve relatively. Looks to be fixed [here](https://github.com/apollographql/apollo-client/commit/77d0bd77a0f935d72fc58e74775d130af0d54bee#diff-aaab2aeb9fe7847f825660d6050709aa)

**Comment by ljharb** [comment]
In that case, closing, since https://github.com/apollographql/apollo-client/commit/77d0bd77a0f935d72fc58e74775d130af0d54bee#diff-aaab2aeb9fe7847f825660d6050709aa fixed it.
